tools/memap.py

```diff
@@ -616,6 +619,67 @@ def generate_table(self, file_desc):
 
         return output
 
+    def generate_bars(self, file_desc, device_name):
+        """ Generates nice looking bars that represent the memory consumption
+
+        Returns: string containing nice looking bars
+        """
+
+        # TODO add tty detection, and width detection probably
```

Comment:
Was this a TODO you wanted to address before merging or is it fine as is?

Reply:
Either or. You can see the current width detection below, which will only work on Linux (but doesn't hurt other OSs):
https://github.com/ARMmbed/mbed-os/blob/a4bfd01f7dd58e066e94ce5d858f03ac853f5ad9/tools/memap.py#L628-L638

I'm not planning on adding support for other OSs at this time, though I don't think that should stop this PR. They're just stuck at a 72 character width. I did search briefly, but unfortunately finding cross-platform screen width is a nightmare in python 2.7.

Reply:
Fair enough.